### PR TITLE
[ADD] pos_minimize_menu : new module to save place to display current ticket

### PR DESCRIPTION
**Pull request by legalsylvain**
This module allows you to minimize the placeholder that contains
all the options in main point of sale menu part.

It allows to have more place for the display of the current ticket.

**Without this module:**

without_module

**with this module:**

with_module

if you want to display in the main screen some important buttons:

* Go to 'Point of Sale > Configuration > Settings'

* Edit the field 'Important Buttons' and write the technical name of the buttons.

configure_important_buttons

Here are for the official modules, the following possible values:

  * ``point_of_sale`` : ProductInfoButton, SetFiscalPositionButton, OrderlineCustomerNoteButton, SetPricelistButton, RefundButton

  * ``pos_sale``: SetSaleOrderButton

  * ``pos_discount``: DiscountButton

  * ``pos_loyalty``: RewardButton, ResetProgramsButton, eWalletButton, PromoCodeButton

  * ``pos_restaurant``: SplitBillButton, SubmitOrderButton, TransferOrderButton, PrintBillButton, TableGuestsButton, OrderlineNoteButton

As a result, selected buttons will be displayed in the main screen:

important_buttons_displayed

### CODE REVIEW

File: pos_minimize_menu/static/src/js/ProductScreen.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    "use strict";
+
+    const ProductScreen = require("point_of_sale.ProductScreen");
+    const Registries = require("point_of_sale.Registries");
+
+    const OverloadProductScreen = (OriginalProductScreen) =>
+        class extends OriginalProductScreen {
+            get importantControlButtons() {
+                var importantButtons = [];
+                console.log("=============================");
```
Comment (nseinlet) [inline pos_minimize_menu/static/src/js/ProductScreen.js]: Is it a good practice to leave the console.log ?

Comment (legalsylvain) [reply pos_minimize_menu/static/src/js/ProductScreen.js]: hehe ! absolutely not ! thanks ! 
Fixed.

File: pos_minimize_menu/models/pos_config.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+class PosConfig(models.Model):
+    _inherit = "pos.config"
+
+    iface_important_buttons = fields.Text(
+        string="Important Buttons",
+        help="Set technical names of buttons that will"
+        " be displayed in the main point of sale screen."
+        " Other buttons will be available by clicking"
+        " on the button 'More...'.",
+        default=True,
```
Comment (nseinlet) [inline pos_minimize_menu/models/pos_config.py]: strange default for a text field

Comment (legalsylvain) [reply pos_minimize_menu/models/pos_config.py]: :+1:, removed.

**Review by hparfr — APPROVED** [review]
lgtm

**Comment by nseinlet** [comment]
Looks a bit too technical to specify button names. Probably better to use an x2many with many2many_tags widget and compute the string to send to pos JS.

**Comment by legalsylvain** [comment]
> Looks a bit too technical to specify button names. Probably better to use an x2many with many2many_tags widget and compute the string to send to pos JS.

Hi. yes I thought to that solution. but it is not extendable. Button are introduced by many modules. point_of_sale, pos_restaurant, etc... Can not now the list in the back office, AFAIK. Don't you think ? 

thanks !

**Review by PaulGoubert — APPROVED** [review]
Functionnal review OK.   The module work well and hide all the option below one "more" button.     
Thank you @legalsylvain

**Review by chienandalu — APPROVED** [review]
Cool @legalsylvain :+1: 

/ocabot merge nobump

**Pull request merged**